### DISCUSSION TITLE
Option to add PR number to "Rebase and Merge" commits

### DISCUSSION THREAD

**Discussion by IanButterworth** [community/community#62118] [Pull Requests]
### Select Topic Area

Product Feedback

### Body

Currently "Rebase and Merge" doesn't meet the needs of a project I am part of because the commits cannot easily be tracked back to the PR.

It would be very helpful to be able to trace commits from a "Rebase and Merge" from a PR back to the PR.

Perhaps a repo-enableable option could be added to automatically add a PR reference to the commits?

Say, turn these commits from PR #123
`commit 1`
`commit 2`
into
`commit 1 (#123)`
`commit 2 (#123)`

**Comment by mujeeb00** [comment]
Product Feedback:

Thank you for providing valuable feedback regarding the "Rebase and Merge" feature. We understand the importance of maintaining clear traceability of commits back to their respective pull requests, and your suggestion to automatically add a PR reference to the commits is a great idea to address this concern.

Having the ability to trace commits back to the PR would greatly enhance project organization and make it easier for contributors and maintainers to understand the history and context of each commit. This feature could prove to be particularly beneficial for projects with a significant number of pull requests and contributors.

To implement this feature, we can consider introducing an option that can be enabled at the repository level. When this option is enabled, the "Rebase and Merge" action would automatically add a reference to the corresponding pull request in the commit message. For example, the commit messages would be updated as follows:

before commit 1
commit 2
after
commit 1 (#123)
commit 2 (#123)

This way, contributors and maintainers can easily identify which pull request a commit belongs to, and it would improve the overall clarity and transparency of the project's commit history.

We appreciate your valuable input in helping us improve our platform. Our development team will carefully consider your feedback, and we will explore the feasibility of implementing this feature. As we continually strive to enhance GitHub's functionality, suggestions like yours play a crucial role in shaping the future of our platform.

If you have any further feedback, questions, or suggestions, please don't hesitate to share them with us. We value your input and look forward to making GitHub an even better place for collaboration and version control. Thank you for being a part of our community!

**Comment by mdonatas-trafi** [comment]
Right now looking at `Rebase and Merge` commits one may get an impression that someone simply force pushed to `main`. Additionally it's not possible to search for merge commits by PR number in git log / commit tree in git software.

**Reply by IanButterworth** [reply]
Absolutely. It throws away the link back to the PR so is basically unusable for large multi contributor repos.

**Comment by LaurenceKiln** [comment]
Please fix this. Commits cannot be undone. the longer this issue remains unaddressed the more  project discussion becomes forever disconnected from the commit history, for every project on github that uses this feature. :angry:
